src/actions/jira/jira.ts

```diff
@@ -1,133 +1,212 @@
-          `\nLooker URL: ${request.scheduledPlan && request.scheduledPlan.url}`,
-        issuetype: {
-          id: request.formParams.issueType,
-        },
+      project: {
+        id: request.formParams.project!,
+      },
+      summary: request.formParams.summary,
+      description: request.formParams.description,
+      url,
+      issuetype: {
+        id: request.formParams.issueType!,
       },
     }
-    let response
-    try {
-      await jira.addNewIssue(issue)
-    } catch (e) {
-      response = {success: false, message: e.message}
+
+    const stateJson = JSON.parse(request.params.state_json)
+    if (stateJson.tokens && stateJson.redirect) {
+      try {
+        const client = await this.jiraClient(stateJson.redirect, stateJson.tokens)
+        const newIssue = await client.newIssue(issue)
+        await client.addAttachmentToIssue(newIssue.key, buffer, filename, request.attachment.mime)
+        resp.success = true
+      } catch (e) {
+        resp.success = false
+        resp.message = e.message
+      }
+    } else {
+      resp.success = false
+      resp.state = new Hub.ActionState()
+      resp.state.data = "reset"
     }
-    return new Hub.ActionResponse(response)
+    return new Hub.ActionResponse(resp)
   }
 
   async form(request: Hub.ActionRequest) {
+    if (request.params.state_json) {
+      try {
+        const stateJson = JSON.parse(request.params.state_json)
+        if (stateJson.tokens && stateJson.redirect) {
+          const client = await this.jiraClient(stateJson.redirect, stateJson.tokens)
+          const projects = await client.getProjects()
+          const projectOptions: { name: string, label: string }[] = projects.map((p: any) => {
+            return { name: p.id, label: p.name }
+          })
 
-    const form = new Hub.ActionForm()
-    try {
-      const jira = this.jiraClientFromRequest(request)
-
-      const [projects, issueTypes] = await Promise.all([
-        jira.listProjects(),
-        jira.listIssueTypes(),
-      ])
-
-      form.fields = [{
-        default: projects[0].id,
-        label: "Project",
-        name: "project",
-        options: projects.map((p: any) => {
-          return {name: p.id, label: p.name}
-        }),
-        type: "select",
-        required: true,
-      }, {
-        label: "Summary",
-        name: "summary",
-        type: "string",
-        required: true,
-      }, {
-        label: "Description",
-        name: "description",
-        type: "textarea",
-        required: true,
-      }, {
-        default: issueTypes[0].id,
-        label: "Issue Type",
-        name: "issueType",
-        type: "select",
-        options: issueTypes
-          .filter((i: any) => i.description)
-          .map((p: any) => {
-            return {name: p.id, label: p.name}
-          }),
-        required: true,
-      }]
-    } catch (e) {
-      form.error = e
+          const issueTypesOptions = [{
```

Comment:
I've tested this for all 5 issue types. Bug, Story, and Task work, but I'm getting the following errors for Epics and Sub-tasks:

`400 - {"errorMessages":[],"errors":{"customfield_10011":"Epic Name is required."}}`	
`400 - {"errorMessages":[],"errors":{"issuetype":"Issue type is a sub-task but parent issue key or id not specified."}}`

Could we add an interactive form fields for the epic name and sub-task if the user selects one of these?

Reply:
For sub-tasks, we should be able to write a jql query to return any possible issue ids